include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -5367,6 +5384,521 @@ struct OrtApi {
                   _In_ uint32_t vendor_id, _In_ int16_t device_id, _In_ enum OrtDeviceMemoryType mem_type,
                   _In_ size_t alignment, enum OrtAllocatorType allocator_type,
                   _Outptr_ OrtMemoryInfo** out);
+
+  //
+  // OrtArrayOfConstObjects
```

Comment:
is the intended usage of this limited to allowing APIs to return a container of read only things? do we need to expose ways to construct the container and modify it in the public API?

Reply:
We'll need the APIs to construct and add things in future PRs, like `IExecutionProvider::GetEpContextNodes()`. I'm currently working on that right now, and I construct a `OrtArrayOfConstObjects`, add `const OrtNode*` elements to it, and return it from `GetEpContextNodes`.

I probably should have waited to add these in that future PR...